### PR TITLE
feat(source/mysql): Upgrade to `github.com/cloudquery/plugin-sdk/v2` 

### LINKED ISSUE #10034: feat: Migrate `plugin/source/mysql` to `github.com/cloudquery/plugin-sdk/v2`
### Which problem is this feature request solving?

Part of #9846

### Describe the solution you'd like

Upgrade the code.

### Pull request (optional)

- [X] I can submit a pull request

### PR DESCRIPTION
**Pull request by candiduslynx**
Closes #10034.

It includes some nontrivial code for transformation (that is used solely for tests)

### CODE REVIEW

File: plugins/source/mysql/client/transformer.go [hunk 1]
```diff
@@ -135,3 +140,41 @@ func (*Transformer) TransformTextArray(v *schema.TextArray) any {
...
+		}
+		// special handling for arrays
+		switch table.Columns[i].Type {
+		case schema.TypeStringArray,
+			schema.TypeIntArray,
+			schema.TypeUUIDArray,
+			schema.TypeInetArray,
+			schema.TypeCIDRArray,
+			schema.TypeMacAddrArray:
+			parts := strings.Split(strings.TrimSuffix(strings.TrimPrefix(val.(string), "["), "]"), ",")
```
Comment (disq) [inline plugins/source/mysql/client/transformer.go:159]: I wonder if there's a better way to read/deserialize array types from MySQL so we don't have to do this.

Comment (candiduslynx) [reply plugins/source/mysql/client/transformer.go:159]: me too, actually.
Still, I'd like to have @erezrokah to review this, too

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Given the upgrade is non-trivial I suggest waiting until we rolled out destinations so we can review destination first, learn from that and then move to sources.

File: plugins/source/mysql/client/transformer.go (hunk 1 above)
Comment (candiduslynx) [reply plugins/source/mysql/client/transformer.go:159]: I think the issue is still caused only by tests, so the code is safe to go

**Comment by candiduslynx** [comment]
@yevgenypats the non-trivial part is caused only by testing, so the mainline code update was, in fact, trivial.
I think that the testing updates are safe ATM, as the proper update for the types will be done later (proper source plugins Arrow migration).

File: plugins/source/mysql/client/transformer.go [hunk 2]
```diff
@@ -135,3 +140,41 @@ func (*Transformer) TransformTextArray(v *schema.TextArray) any {
 
 	return v.String()
 }
+
+func (*Transformer) RecordToCQTypes(table *schema.Table, record arrow.Record) (schema.CQTypes, error) {
```
Comment (yevgenypats) [inline plugins/source/mysql/client/transformer.go:144]: Let's just bring back the old `testData` that generates CQTypes directly. this way this migration will be much easier (same as in oracleDB)

**Review by yevgenypats — CHANGES_REQUESTED** [review]

File: plugins/source/mysql/client/transformer.go (hunk 2 above)
Comment (candiduslynx) [reply plugins/source/mysql/client/transformer.go:144]: https://github.com/cloudquery/plugin-sdk/pull/811

Comment (candiduslynx) [reply plugins/source/mysql/client/transformer.go]: Done in a8c4fb63be7ed5a1c855e3125466a8cda2be0917

**Review by yevgenypats — APPROVED** [review]

**Review by erezrokah — APPROVED** [review]

**Pull request merged**